### PR TITLE
force compound to be grouped together in chemical formula

### PR DESCRIPTION
**Pull request by Glease**
fix https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/12187

### CODE REVIEW

**Comment by miozune** [comment]
This doesn't look right

File: src/main/java/gregtech/api/objects/MaterialStack.java [hunk 1]
```diff
@@ -43,14 +43,24 @@ public String toString() {
...
             temp4 = GT_Utility.toSubscript(mAmount);
+        }
+        if (mMaterial.mMaterialList.size() > 1 || isMaterialListComplex(this)) {
+            temp1 = "(";
+            temp3 = ")";
+        }
+        return temp1 + temp2 + temp3 + temp4;
+    }
 
+    public String toStringSingle() {
```
Comment (miozune) [inline src/main/java/gregtech/api/objects/MaterialStack.java:54]: Code duplication can be reduced here. Add boolean parameter `parenthesizeSingle` or smth and merge two methods

File: src/main/java/gregtech/api/enums/Materials.java [hunk 2]
```diff
@@ -1467,10 +1467,12 @@ public Materials(
...
         mExtraData = aExtraData;
         mMaterialList.addAll(aMaterialList);
-        mChemicalFormula = mMaterialList.stream()
-                .map(MaterialStack::toString)
-                .collect(Collectors.joining())
-                .replaceAll("_", "-");
+        if (mMaterialList.size() == 1) mChemicalFormula = mMaterialList.get(0).toStringSingle();
+        else
+            mChemicalFormula = mMaterialList.stream()
+                    .map(MaterialStack::toString)
```
Comment (miozune) [inline src/main/java/gregtech/api/enums/Materials.java:1473]: `mMaterialList.size() == 1 ? _method1_ : _method2_` might be better

Comment (Glease) [reply src/main/java/gregtech/api/enums/Materials.java:1473]: I'd prefer to not use ternary expression when long expressions are involved. Readability goes down the drain that way.

File: src/main/java/gregtech/api/objects/MaterialStack.java (hunk 1 above)
Comment (Glease) [reply src/main/java/gregtech/api/objects/MaterialStack.java:54]:
```java
    public String toString() {
        return toString(false);
    }

    public String toString(boolean single) {
        String temp1 = "", temp2 = mMaterial.getToolTip(true), temp3 = "", temp4 = "";
        if (mAmount > 1) {
            temp4 = GT_Utility.toSubscript(mAmount);
        }
        if ((!single || mAmount > 1) && (mMaterial.mMaterialList.size() > 1 || isMaterialListComplex(this))) {
            temp1 = "(";
            temp3 = ")";
        }
        return temp1 + temp2 + temp3 + temp4;
    }
```
not sure if I can still understand what this mess mean when I wake up tomorrow.

File: src/main/java/gregtech/api/enums/Materials.java (hunk 2 above)
Comment (miozune) [reply src/main/java/gregtech/api/enums/Materials.java:1473]: I see

File: src/main/java/gregtech/api/objects/MaterialStack.java (hunk 1 above)
Comment (Glease) [reply src/main/java/gregtech/api/objects/MaterialStack.java]: it does look a bit better.

**Review by miozune — APPROVED** [review]

**Pull request merged**